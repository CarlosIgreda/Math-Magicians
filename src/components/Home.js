import { Link } from 'react-router-dom';

function Home() {
  const characterCount = 500;
  const lorem = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam aliquam consectetur felis nec commodo. Integer consectetur sed neque nec ultrices. Sed facilisis tristique ligula, ut consectetur libero posuere nec. Duis fringilla nulla ac aliquet rutrum. Maecenas eu elit at neque dapibus mollis. Aliquam erat volutpat. Morbi quis pulvinar purus. Sed sagittis quam id nisl fermentum, a scelerisque felis fermentum. Proin imperdiet ante at lorem finibus, non posuere arcu condimentum. Etiam eu lorem justo. Morbi rutrum felis vel tempor varius. Aliquam erat volutpat. Morbi quis felis orci. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Fusce auctor odio id libero consectetur, vitae fermentum mi tristique.';

  const limitedLorem = lorem.substring(0, characterCount);

  return (
    <div>
      <nav className="header">
        <h1 className="title">üìê Math Magiciansüìè</h1>
        <ul className="menu">
          <li className="menu-item">
            <Link to="/Math-Magicians">Home</Link>
          </li>
          <li className="menu-item">
            <Link to="/Math-Magicians/calculator">Calculator</Link>
          </li>
          <li className="menu-item">
            <Link to="/Math-Magicians/quote">Quote</Link>
          </li>
        </ul>
      </nav>
      <section className="home-container" data-testid="lorem-container">
        <h1 className="home-title">Welcome to our page!</h1>
        <p className="lorem-text">{limitedLorem}</p>
        <p className="lorem-text">{limitedLorem}</p>
      </section>
    </div>
  );
}

export default Home;
