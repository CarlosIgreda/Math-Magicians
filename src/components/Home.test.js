import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Home from './Home';

describe('Home', () => {
  it('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
