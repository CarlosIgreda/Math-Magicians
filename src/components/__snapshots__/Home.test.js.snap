// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div>
  <div>
    <nav
      class="header"
    >
      <h1
        class="title"
      >
        üìê Math Magiciansüìè
      </h1>
      <ul
        class="menu"
      >
        <li
          class="menu-item"
        >
          <a
            href="/Math-Magicians"
          >
            Home
          </a>
        </li>
        <li
          class="menu-item"
        >
          <a
            href="/Math-Magicians/calculator"
          >
            Calculator
          </a>
        </li>
        <li
          class="menu-item"
        >
          <a
            href="/Math-Magicians/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
    <section
      class="home-container"
    >
      <h1
        class="home-title"
      >
        Welcome to our page!
      </h1>
      <p
        class="lorem-text"
      >
        Eu est tempor aliquip elit cupidatat aliqua voluptate. Nulla occaecat incididunt ipsum commodo qui aliqua fugiat duis occaecat ea non dolor nostrud. Duis cupidatat pariatur reprehenderit consectetur exercitation mollit aliqua labore pariatur duis. Reprehenderit magna laborum qui Lorem. Nisi magna voluptate tempor commodo in deserunt excepteur dolore sint tempor. Ipsum duis irure irure reprehenderit dolor sit. Consequat aute aliquip proident aute adipisicing est. Fugiat tempor ut consectetur est. Est aute incididunt occaecat ex incididunt. Esse eiusmod culpa ea amet.
      </p>
      <p
        class="lorem-text"
      >
        Eu est tempor aliquip elit cupidatat aliqua voluptate. Nulla occaecat incididunt ipsum commodo qui aliqua fugiat duis occaecat ea non dolor nostrud. Duis cupidatat pariatur reprehenderit consectetur exercitation mollit aliqua labore pariatur duis. Reprehenderit magna laborum qui Lorem. Nisi magna voluptate tempor commodo in deserunt excepteur dolore sint tempor. Ipsum duis irure irure reprehenderit dolor sit. Consequat aute aliquip proident aute adipisicing est. Fugiat tempor ut consectetur est. Est aute incididunt occaecat ex incididunt. Esse eiusmod culpa ea amet.
      </p>
    </section>
  </div>
</div>
`;
